ci/generate_azure_pipelines_matrices.py

```diff
@@ -201,10 +201,15 @@ def __init__(self, platform, build_jobs):
             subarchives="qtbase qttools qt icu",
         ),
         BuildJob(
-            "install-src", "6.1.0", "linux", "desktop", "gcc_64", "gcc_64", subarchives="qt"
+            "install-src", "6.1.0", "linux", "desktop", "gcc_64", "gcc_64", subarchives="qtlottie"
         ),
         BuildJob(
-            "install-doc", "6.1.0", "linux", "desktop", "gcc_64", "gcc_64", subarchives="qtdoc"
+            "install-doc", "6.1.0", "linux", "desktop", "gcc_64", "gcc_64",
+            subarchives="qtdoc qtlottieanimation", module="qtlottie"
+        ),
+        BuildJob(
+            "install-example", "6.1.0", "linux", "desktop", "gcc_64", "gcc_64",
+            subarchives="qtdoc qtlottie", module="qtlottie"
```

Comment:
This code illustrates a potential problem with `list-doc`/`list-example`: These commands cannot list archives for modules. This is a problem for the `qtlottie` module, because the archive is called `qtlottieanimation` when you run `install-doc`, but it's called `qtlottie` everywhere else.

We could add a function that prints the names of archives for each module, the way we did for `list-qt`. I don't like this idea, because I think it would be better to just make sure that modules are installed by default whenever you ask for them. I'll open an issue to track this; I prefer to solve this problem in a separate PR.

Reply:
**Please note:** The module `qtlottie` is a bad choice for `install-example` with Qt 6.1.0; the archive is empty!

```console
$ 7z l 6.1.0-0-202105040937qtlottie-examples-6.1.0.7z

7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=C.UTF-8,Utf16=on,HugeFiles=on,64 bits,6 CPUs Intel(R) Core(TM) i5-8600K CPU @ 3.60GHz (906EA),ASM,AES-NI)

Scanning the drive for archives:
1 file, 162 bytes (1 KiB)

Listing archive: /mnt/d/Downloads/6.1.0-0-202105040937qtlottie-examples-6.1.0.7z

--
Path = /mnt/d/Downloads/6.1.0-0-202105040937qtlottie-examples-6.1.0.7z
Type = 7z
Physical Size = 162
Headers Size = 162
Solid = -
Blocks = 0

   Date      Time    Attr         Size   Compressed  Name
------------------- ----- ------------ ------------  ------------------------
2021-05-04 01:37:31 D....            0            0  Examples
2021-05-04 01:43:19 D....            0            0  Examples/Qt-6.1.0
------------------- ----- ------------ ------------  ------------------------
2021-05-04 01:43:19                  0            0  0 files, 2 folders
```
I chose this module because of the archive's small file size; I didn't realize there wasn't anything in it!

I should have seen this in Updates.xml:
```xml
<PackageUpdate>
  <Name>qt.qt6.610.examples.qtlottie</Name>
...
  <UpdateFile OS="Any" CompressedSize="202" UncompressedSize="40"/>
...
</PackageUpdate>
```

Maybe we could print a warning when `PackageUpdate.UpdateFile.CompressedSize` is greater than `PackageUpdate.UpdateFile.UncompressedSize`, or when UncompressedSize is less than some threshold value? It could be something like "This module is only 40 bytes when uncompressed; it might be empty!"

Reply:
`qt.qt6.610.examples.qtimageformats` has this issue as well.

Reply:
Resolved in d6a6f79: this prevents the `list-*` commands from printing any module <= 40 bytes when decompressed. The 40 bytes value can be adjusted in `settings.ini`. I don't think this will be a popular thing to change, so I will not make a CLI flag for it.

I did not change the behavior of `install-*` for this. You can still install empty modules, but you won't see them with `list-*` unless you change your settings.

This will cause `combinations.json` to omit empty modules, which means that `install-*` will warn the user that the empty modules are unrecognized.

Reply:
> any module <= 40 bytes when decompressed. 

It's nice findings how to detect empty modules.